maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/ConcorContributionsService.java

```diff
@@ -61,30 +64,36 @@ public List<ConcorContributionResponse> getConcorContributionFiles(ConcorContrib
     }
 
     @Transactional(rollbackFor = MAATCourtDataException.class)
-    public boolean createContributionAndUpdateConcorStatus(CreateContributionFileRequest contributionRequest){
+    @NotNull
+    public Integer createContributionAndUpdateConcorStatus(CreateContributionFileRequest contributionRequest){
 
         ValidationUtils.isNull(contributionRequest,"contributionRequest object is null");
         ValidationUtils.isEmptyOrHasNullElement(contributionRequest.getConcorContributionIds(),"ContributionIds are empty/null.");
 
         final ContributionFilesEntity contributionFilesEntity = createContributionFile(contributionRequest);
-        return updateConcorContributionStatusAndSetContribFile(contributionRequest.getConcorContributionIds(), SENT, contributionFilesEntity.getFileId());
+        if (!updateConcorContributionStatusAndSetContribFile(contributionRequest.getConcorContributionIds(), SENT, contributionFilesEntity.getFileId())) {
+            throw new NoSuchElementException("No concor_contribution status values were updated"); // did not rollback previously
+        }
+        if (contributionFilesEntity.getFileId() == null) {
+            throw new RequestedObjectNotFoundException("Created contribution_file's id could not be found"); // did not rollback previously
+        }
+        return contributionFilesEntity.getFileId();
     }
 
     @Transactional(rollbackFor =  MAATCourtDataException.class)
-    public boolean logContributionProcessed(LogContributionProcessedRequest request){
-        boolean successful = false;
-        Optional<ConcorContributionsEntity> optionalConcorEntry = concorRepository.findById(request.getConcorId());
-        if(optionalConcorEntry.isPresent()) {
-            ConcorContributionsEntity concorEntity = optionalConcorEntry.get();
-            log.info("Contribution found: {}", concorEntity.getId());
-            successful = debtCollectionService.updateContributionFileReceivedCount(concorEntity.getContribFileId());
-            // check if error
-            if(successful && !StringUtils.isEmpty(request.getErrorText()) ){
-                successful = saveErrorMessage(request, concorEntity);
-            }
-
+    @NotNull
+    public Integer logContributionProcessed(LogContributionProcessedRequest request){
+        ConcorContributionsEntity concorEntity = concorRepository.findById(request.getConcorId())
+                .orElseThrow(() -> new RequestedObjectNotFoundException("concor_contribution could not be found by id"));
+        log.info("Contribution found: {}", concorEntity.getId());
+        if (!debtCollectionService.updateContributionFileReceivedCount(concorEntity.getContribFileId())) {
+            throw new NoSuchElementException("No associated contribution_file found for concur_contribution");
         }
-        return successful;
+        if(!StringUtils.isEmpty(request.getErrorText()) ){
+            saveErrorMessage(request, concorEntity);
```

Comment:
Never returns `false` (it will return `true` or throw an exception)

Reply:
> If it'll only ever return true, or exception, is it worth pulling the boolean and swapping to void?

@LAA-Dave-Edwards I did do that in the original patch :), but the patch started to become quite big. Was trying to keep the PR focussed on just the essential change. Let me know if I should change it? Otherwise I can keep a small list of 'cleanup things' and submit myself a cleanup ticket at the end of the week?